compiler/codegen/CodeGenRA.cpp

```diff
@@ -458,24 +458,36 @@ OMR::CodeGenerator::allocateSpill(int32_t dataSize, bool containsCollectedRefere
    //
    bool try8ByteSpills = (dataSize < 16) && ((TR::Compiler->om.sizeofReferenceAddress() == 8) || !containsCollectedReference);
    bool try16ByteSpills = (dataSize == 16);
+   bool try32ByteSpills = (dataSize == 32);
+   bool try64ByteSpills = (dataSize == 64);
 
    if(reuse)
-     {
-     if (dataSize <= 4 && !self()->getSpill4FreeList().empty())
-     {
-       spill = self()->getSpill4FreeList().front();
-       self()->getSpill4FreeList().pop_front();
-     }
-     if (!spill && try8ByteSpills && !self()->getSpill8FreeList().empty())
-     {
-       spill = self()->getSpill8FreeList().front();
-       self()->getSpill8FreeList().pop_front();
-     }
-     if (!spill && try16ByteSpills && !self()->getSpill16FreeList().empty())
-     {
-       spill = self()->getSpill16FreeList().front();
-       self()->getSpill16FreeList().pop_front();
-     }
+      {
+      if (dataSize <= 4 && !self()->getSpill4FreeList().empty())
+         {
+         spill = self()->getSpill4FreeList().front();
+         self()->getSpill4FreeList().pop_front();
+         }
+      if (!spill && try8ByteSpills && !self()->getSpill8FreeList().empty())
```

Comment:
Shouldn't all these `if`s be `else if`s instead?  I think only one of these cases would ever apply at any given moment.